libcontainer/seccomp/patchbpf/enosys_linux.go

```diff
@@ -114,14 +115,26 @@ func disassembleFilter(filter *libseccomp.ScmpFilter) ([]bpf.Instruction, error)
 	defer wtr.Close()
 	defer rdr.Close()
 
+	readerBuffer := new(bytes.Buffer)
+	errChan := make(chan error)
+	go func() {
+		_, err := io.Copy(readerBuffer, rdr)
+		errChan <- err
```

Comment:
This will block forever if the `filter.ExportBPF` below returns an error, so no one would read from this channel.

The simplest solution is to make the channel buffered, i.e. `errChan := make(chan error, 1)` above.

Reply:
> This will block forever

... resulting in a stuck goroutine